src/optional.h

```diff
@@ -157,6 +157,14 @@ class optional
             }
         }
 
+        void clear() {
+            reset();
+        }
+
+        bool empty() const {
+            return !has_value();
```

Comment:
Same here. This class is modeled after `std::optional`, which has no `empty()` member, please don't add this and use  `has_value` directly.